src/internal/utils/bucketHelper.ts

```diff
@@ -43,19 +44,26 @@ import { UICONFIG } from "../constants/uiConfig";
 import { getTranslation } from "./translations";
 
 export function sanitizeFilters(newReferencePoint: IExtendedReferencePoint): IExtendedReferencePoint {
-    const usedAttributeBucketItems = getAllAttributeItems(newReferencePoint.buckets);
+    const attributeBucketItems = getAllAttributeItems(newReferencePoint.buckets);
+    const measureBucketItems = getAllMeasureItems(newReferencePoint.buckets);
 
     const filteredFilters = newReferencePoint.filters.items.filter((filterBucketItem: IFiltersBucketItem) => {
-        const hasCorrespondingBucketItem = usedAttributeBucketItems.some(
-            (attributeBucketItem: IBucketItem) => {
-                const filter = filterBucketItem.filters[0];
-                if (isAttributeFilter(filter) || isDateFilter(filter)) {
-                    return attributeBucketItem.attribute === filter.attribute;
-                }
-                return false;
-            },
-        );
-        return filterBucketItem.autoCreated === false || hasCorrespondingBucketItem;
+        const filter = filterBucketItem.filters[0];
+
+        if (isAttributeFilter(filter) || isDateFilter(filter)) {
+            const hasCorrespondingBucketItem = attributeBucketItems.some(
+                (attributeBucketItem: IBucketItem) => attributeBucketItem.attribute === filter.attribute,
+            );
+            return filterBucketItem.autoCreated === false || hasCorrespondingBucketItem;
+        } else if (isMeasureValueFilter) {
+            const hasCorrespondingBucketItem = measureBucketItems.some(
+                (measureBucketItem: IBucketItem) =>
+                    measureBucketItem.localIdentifier === filter.measureLocalIdentifier,
+            );
+            return attributeBucketItems.length !== 0 && hasCorrespondingBucketItem;
+        }
+
+        return false;
```

Comment:
How about changing the processing in the following way? It would be a bit more readable and save some computation (especially for auto-created filters):

````
    const filteredFilters = newReferencePoint.filters.items.filter((filterBucketItem: IFiltersBucketItem) => {
        const filter = filterBucketItem.filters[0];

        if (isAttributeFilter(filter) || isDateFilter(filter)) {
            if (filterBucketItem.autoCreated === false) {
                return true;
            }
            return attributeBucketItems.some((attributeBucketItem: IBucketItem) => attributeBucketItem.attribute === filter.attribute);
        } else if (isMeasureValueFilter) {
            if (attributeBucketItems.length > 0) {
                return true;
            }
            return measureBucketItems.some((measureBucketItem: IBucketItem) => measureBucketItem.localIdentifier === filter.measureLocalIdentifier);
        }

        return false;
    });
````

Reply:
Good idea. I've done it as suggested here, just the second one has AND condition so it needed some slight changes in the logic (`return false` if not)